src/basic/namespace-util.c

```diff
@@ -95,7 +95,7 @@ int namespace_open(
                 const char *root;
 
                 root = procfs_file_alloca(pid, "root");
-                root_fd = open(root, O_RDONLY|O_NOCTTY|O_CLOEXEC|O_DIRECTORY);
+                root_fd = open(root, O_PATH|O_CLOEXEC|O_DIRECTORY);
```

Comment:
I'm really against introducing potential footgun to a generic helper function like this. `namespace_open()` is not on any performance critical path, and it would be inconsistent here that only root_fd is O_PATH.